Mathlib/Algebra/Category/Ring/Topology.lean

```diff
@@ -0,0 +1,176 @@
+/-
+Copyright (c) 2025 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang, Christian Merten
+-/
+import Mathlib.Algebra.Category.Ring.Colimits
+import Mathlib.Algebra.Category.Ring.Constructions
+import Mathlib.Algebra.MvPolynomial.CommRing
+import Mathlib.Topology.Algebra.Ring.Basic
+
+/-!
+# Topology on `Hom(R, S)`
+
+In this file, we define topology on `Hom(R, S)` for a topological ring `S`,
+given by the coarsest topology that makes `f ↦ f x` continuous for all `x : R`.
+Alternatively, given a presentation `R = ℤ[x₁,...,xₙ]/I`,
```

Comment:
Maybe "For finitely-generated rings `R = ℤ[x₁,...,xₙ]/I` this can alternatively be
described as the subspace..."?